### ISSUE TITLE
TY the Tasmanian Tiger [45410029]

### ISSUE THREAD

**Issue by jeltaqq** [Cxbx-Reloaded/game-compatibility#583]
### Description

Booting, but when goto title screen has crash.

```
Received Exception Code C0000005 @ EIP := 67FB7139
```

### Dumps 

[Xbe.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/1648557/Xbe.txt)
[KrnlDebug.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/1648558/KrnlDebug.txt)

### Status
**Previous behavior:** 

None reported

**Current behavior:**

state-booting

### Screenshots 
![intro](https://user-images.githubusercontent.com/25340049/35179889-137c5f88-fde7-11e7-82ee-a56b202e4b1e.jpg)

### Additional Information

Tested with 8a63e009

**Comment by ObiKKa** [comment]
This game needs fresh testing on the current Master branch to confirm if it does get in-game.

Shows black screen where there could be intro splash screens.
In-game graphics performed very well in the video below. It may be slightly dark and there's a constant blue-ish smoke filter covering the screen during gameplay. Occasional opaque black bars flickering at the top & bottom of screen in some cutscenes.
Epilespy warning: Black screen flickering early in the video.
- [Cxbx-Reloaded Xbox Emulator - Ty the Tasmanian Tiger Ingame! (090663d7 + WIP)](https://www.youtube.com/watch?v=MstoxWJ-hMM&list=PL4PD8cV5ypNDPTJbrol7A-5rvuteHWWYj&index=7&t=0s) (John GodGames Emus. Published on Mar 15, 2018. Has 5 WIPs.)

**Comment by mnadareski** [comment]
Tested with 587fc9da, it is getting in-game. Can confirm that nearly all opening screens are black. There are some odd issues with rendering and the speech is sped up almost comically fast. Attached is a full log of a successful run.
[KrnlDebug.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/1878536/KrnlDebug.txt)

![image](https://user-images.githubusercontent.com/4381583/38346859-8d7a25a0-384c-11e8-992d-e13a0622a260.png)

**Comment by mnadareski** [comment]
Tested again with 15335306. Things that haven't changed are menus not showing items, the speech and music being either sped up or pitched up (now occasionally pickup jingles are pitched down?) (okay, seems like the longer the clip, the slower it is to "speed up"; one longer dialog started with the correct pitch then just... kept going), prerendered cutscenes not playing properly, savepoints working.

[KrnlDebug.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/2027943/KrnlDebug.txt)

"Water overlay" no longer present and the first hints of dust trails from walking are appearing. (The dust trails probably account for the random black flickering during jumps and animations earlier).

![image](https://user-images.githubusercontent.com/4381583/40378908-ab098ebc-5da9-11e8-8a94-541623e1c096.png)

More items are appearing, but "movable" stuff is still missing.

![image](https://user-images.githubusercontent.com/4381583/40378974-d48bf072-5da9-11e8-9603-2ffeff958a67.png)

Shadows exist, but go a bit technicolor.

![image](https://user-images.githubusercontent.com/4381583/40378997-e7ca5200-5da9-11e8-8179-4c2b937deedd.png)

Opals, crates, and pickups now render correctly, as do "NPC" creatures like wallabies and butterflies.

![image](https://user-images.githubusercontent.com/4381583/40379281-9ab22dca-5daa-11e8-8562-53a652172a33.png)

Water does not seem to render at all now, but the entire screen doesn't go dark and the camera remains normal when stepping in it.

![image](https://user-images.githubusercontent.com/4381583/40379365-bb977482-5daa-11e8-9160-4d7dd561388c.png)

Finally captured the "faint outline" of the mission UI elements. (Couldn't capture it, but near NPCs, you also get the flickering, black outline of the exclamation point above your head.)

![image](https://user-images.githubusercontent.com/4381583/40379942-708ef486-5dac-11e8-97ef-4e2048554f98.png)

**Comment by mnadareski** [comment]
As of 2b8ba74a, I have a couple new things:

Opening credits screen has this weird texture over everything (with a similar texture over the demo):

![image](https://user-images.githubusercontent.com/4381583/43362314-0e14f39e-929c-11e8-95ea-ba16d021c0bd.png)
![image](https://user-images.githubusercontent.com/4381583/43362315-13aa72fc-929c-11e8-9191-a9f3b9f9350f.png)

The "Press Start" text finally appears (over a black background) as does the main menu text and save images (this also extends to all in-game menus):

![image](https://user-images.githubusercontent.com/4381583/43362317-1fc65006-929c-11e8-8c9a-3b852c3b675a.png)
![image](https://user-images.githubusercontent.com/4381583/43362320-3164a0f6-929c-11e8-8d99-cf2b9c02d8cd.png)
![image](https://user-images.githubusercontent.com/4381583/43362325-54450bce-929c-11e8-9da9-5c9d069e0870.png)

Prerendered cutscenes now play!

![image](https://user-images.githubusercontent.com/4381583/43362520-e2013cb8-92a0-11e8-9f0a-87f54483d098.png)

Other things that would be too much of a pain to get pictures of:
- Did loading screens render before? I can't remember. But they load now.
[KrnlDebug.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/2238469/KrnlDebug.txt)

- Sun glare is working but Ty's shadow now pulses from white to black
- Dust and leaves no longer look like broken textures (they actually puff up). Additionally, the death "ghost" of enemies now renders and particle effects around portals (though not the portals themselves) do too
- Pawprints in the dust now stay for the right amount of time
- Some additional UI elements can be seen on level load and on collection (like Rainbow Scales, Thunder Eggs, Opals, and Cogs) as well as "objective" elements, like killing the 7 frills now show up with proper opacity.
- In-game cutscene masking is now working (black bars on top and bottom) but with that odd texture, as seen before (Edit: It appears that it's taking the texture from whatever it can, in the case of cutscenes, it's using the texture of a Rainbow Scale... oddly. In a level, it appears whatever is the closest thing to you is what's used for shadows and the like)
- In-game cutscenes now partially render missing objects like the Talisman Machine (but only in the cutscenes)
- In-game text, such as level names on portals and subtitles, now render correctly
- Music and in-game grunts are all playing properly now. Otherwise in-game cutscenes still have high-pitched and slightly fast dialog
- Thunder egg pickup jingle seems to always play slow, Bilby uncaging always seems to play fast, and getting cogs always seems proper... it's a very strange mix (Edit: After playing through a full level, I'm wrong, all of them play at variable speeds, the pattern was just early in the level).

TL;DR - The game has never been more playable, and I honestly just kept going for longer than the KrnlDebug seems to suggest.

Attached is the KrnlDebug, as always: 
[KrnlDebug.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/2238470/KrnlDebug.txt)

**Comment by mnadareski** [comment]
Update since it's been a significant amount of time since I've tested:
- Intro cutscenes now render as close to console as I have seen with no stuttering or notable audios issues
- No notable issues in menus - all items appear properly, all sound effects work as intended, and there are no lag times between switching menus
- After the intro cutscene and loading screen, the following message shows up:
![image](https://user-images.githubusercontent.com/4381583/73982027-5aba6900-48e8-11ea-8759-c3fc91ec2f44.png)
- In-game cutscenes are still a bit laggy, but there are no rendering issues in them anymore. The black bars show up properly and all in-game menus show up as intended
- Relatively low framerate during gameplay, but there are now no issues with the dust effects, footprints, or boomerang effects.

I can add more pictures later, but this is a significant improvement from when this was last tested.

**Comment by mnadareski** [comment]
Once more, it's been a while. A few updates:
- In-game cutscenes no longer lag with the audio, those issues appear to have all cleared up
- Fading animations going into and out of cutscenes are also functioning, not just the black bars themselves
- Framerate issues appear to have cleared up. The game hits 30FPS constantly even with numerous things like grass and particles on screen.

I've only played a few levels including Rainbow Cliffs in order to test this out, but it seems to be on-par with the console experience now across the board. If I find the time to beat the game in-emulator, I'll try to give another update as well, but I'd call this perfect.

**Comment by mnadareski** [comment]
So I've been continuing on with the game to see if I can play it to completion, still on df22c8b:
- Occasionally (and I can't get this to consistently reproduce), the thunder egg counter doesn't show the proper digit *in* the egg. In the below screenshot, there should be a "1" in the egg in the top right.
![image](https://user-images.githubusercontent.com/4381583/120142674-83c4d400-c193-11eb-86b8-7d4948781805.png)
- The audio issue mentioned before seems to apply to destructable objects as well. In Outback Safari, the sound of fences and huts getting smashed doesn't play either

There are no additional messages in the kernel debug at this point, though I will be collecting them each play session until the end in case anything interesting shows up.

**Comment by mnadareski** [comment]
As promised...
![image](https://user-images.githubusercontent.com/4381583/120268709-301dbd80-c25b-11eb-8118-2b2c9fafad89.png)
![image](https://user-images.githubusercontent.com/4381583/120269386-95be7980-c25c-11eb-8a02-a2c9b85f3939.png)
![image](https://user-images.githubusercontent.com/4381583/120269445-b4bd0b80-c25c-11eb-8270-85778bc7dab8.png)

The game is completable from beginning to end (100%-able too) with no game-breaking bugs or anything outside the super tiny incidental things I reported before.

Kernel Debug files from the 3 sessions it took from last to beaten: [KrnlDebug.zip](https://github.com/Cxbx-Reloaded/game-compatibility/files/6573493/KrnlDebug.zip)
